app/src/main-process/app-window.ts

```diff
@@ -145,6 +159,11 @@ export class AppWindow {
 
     this.window.webContents.on('did-finish-load', () => {
       this.window.webContents.setVisualZoomLevelLimits(1, 1)
+      this.window.webContents.send(
+        'zoom-factor-changed',
+        this.window.webContents.zoomFactor
+      )
+      sendWindowStateEvent(this.window, getWindowState(this.window))
```

Comment:
I don't think this is safe as we can't know for certain that the AppStore is set up to receive these events when `did-finish-load` happens, can we? `did-finish-load` is an Electron event that signals that the page has loaded. `renderer-ready` would be a better approach but I think the safest approach would be to request these states from the AppStore as we did before, only asynchronously this time round.

Reply:
I tried the approach of going from the app-store first, but I had other conflicting issue at the time.. so going to try again. Thanks for explaining the timing of 'did-finish-load' - I did test it out and the values were being set as I expected on the 'did-finish-load' but it sounds like that may not be an guaranteed case.